contracts/fee/FeeHandler.sol

```diff
@@ -3,46 +3,215 @@
+        _validateBuybackToken(_getBatchSize(buybackToken), buybackToken);
+
+        for (uint256 i; i < markets.length; i++) {
+            FeeUtils.claimFees(dataStore, eventEmitter, markets[i], buybackToken, address(this));
         }
 
         address receiver = dataStore.getAddress(Keys.FEE_RECEIVER);
+        IERC20(buybackToken).safeTransfer(receiver, IERC20(buybackToken).balanceOf(address(this)));
+    }
+
+    // @dev claim fees in feeToken from the specified markets
+    // @param market the market from which to claim fees
+    // @param feeToken the fee tokens to claim from the market
+    function claimFees(address market, address feeToken) external nonReentrant {
+        if (_getBatchSize(feeToken) != 0) {
+            revert Errors.InvalidFeeToken(feeToken);
+        }
+
+        uint256 feeAmount;
+        if (market == address(0)) {
+            feeAmount = IVaultGovV1(vaultV1.gov()).withdrawFees(address(vaultV1), feeToken, address(this));
+            _incrementAvailableFeeAmounts(1, feeToken, feeAmount);
+        } else {
+            feeAmount = FeeUtils.claimFees(dataStore, eventEmitter, market, feeToken, address(this));
+            _incrementAvailableFeeAmounts(2, feeToken, feeAmount);
+        }
+    }
+
+    // @dev receive an amount in feeToken by depositing the batchSize amount of the buybackToken
+    // @param feeToken the token to receive with the fee amount calculated via an oracle price
+    // @param buybackToken the token to deposit in the amount of batchSize in return for fees
+    // @param minOutputAmount the minimum amount of the feeToken that the caller will receive
+    function buyback(address feeToken, address buybackToken, uint256 minOutputAmount) external nonReentrant {
+        if (feeToken == buybackToken) {
+            revert Errors.BuybackAndFeeTokenAreEqual(feeToken, buybackToken);
+        }
+
+        uint256 batchSize = _getBatchSize(buybackToken);
+        _validateBuybackToken(batchSize, buybackToken);
 
+        uint256 availableFeeAmount = _getAvailableFeeAmount(feeToken, buybackToken);
+        if (availableFeeAmount == 0) {
+            revert Errors.AvailableFeeAmountIsZero(feeToken, buybackToken, availableFeeAmount);
+        }
+
+        uint256 maxFeeTokenAmount = _getMaxFeeTokenAmount(feeToken, buybackToken, batchSize);
+        uint256 outputAmount = availableFeeAmount < maxFeeTokenAmount ? availableFeeAmount : maxFeeTokenAmount;
+
+        if (outputAmount < minOutputAmount) {
+            revert Errors.InsufficientBuybackOutputAmount(feeToken, buybackToken, outputAmount, minOutputAmount);
+        }
+
+        _buybackFees(feeToken, buybackToken, batchSize, outputAmount, availableFeeAmount);
+    }
+
+    function getOutputAmount(
+        address[] calldata markets,
+        address feeToken,
+        address buybackToken
+    ) external view returns (uint256) {
+        uint256 batchSize = _getBatchSize(buybackToken);
+        _validateBuybackToken(batchSize, buybackToken);
```

Comment:
small suggestion, i think we usually don't use prefix "_" for private/internal methods, probably makes sense not to use it here for consistency?

Reply:
we do use prefix "_" for private / internal methods: https://github.com/gmx-io/gmx-synthetics/blob/main/contracts/exchange/OrderHandler.sol#L352

i think only in libraries do we not do that

Reply:
i see, i had an impression that this prefix is an exception and rarely used